File: app/models/context.py
from typing import List, Optional
from pydantic import BaseModel

class ClientContext(BaseModel):
    client_name: Optional[str] = None
    industry: Optional[str] = None
    location: Optional[str] = None
    engagement_age: int = 0  # in years
    business_overview: Optional[str] = None
    objectives: List[str] = []
    company_info: Optional[str] = None
    additional_context_questions: List[str] = []
    potential_future_opportunities: List[str] = []

class AnalyzeRequest(BaseModel):
    client_name: Optional[str] = None
    raw_text_blocks: List[str] = []

class AnalyzeResponse(BaseModel):
    analysis_id: str
    status: str
    summary: ClientContext

File: frontend/src/types/context.ts
// Type definitions aligned with backend app.models.context.ClientContext
export interface ClientContext {
  client_name: string | null;
  industry: string | null;
  location: string | null;
  engagement_age: number; // years
  business_overview: string | null;
  objectives: string[];
  company_info: string | null;
  additional_context_questions: string[];
  potential_future_opportunities: string[];
  // The backend may include evidence pointers or notes in future; keep index signature optional
  [key: string]: unknown;
}

export interface AnalyzeResponse {
  analysis_id: string;
  status: string; // "completed" | "error" | other future states
  summary: ClientContext;
  // Backend might send additional keys (e.g., log) later
  [key: string]: unknown;
}
